packages/js/test/plugins/auctionHouseModule/executeSale.test.ts

```diff
@@ -0,0 +1,50 @@
+import test, { Test } from 'tape';
+import { sol, toPublicKey } from '@/types';
+import {
+  metaplex,
+  killStuckProcess,
+  createNft,
+  createWallet,
+} from '../../helpers';
+import { createAuctionHouse } from './helpers';
+
+killStuckProcess();
+
+test('[auctionHouseModule] execute sale on an Auction House', async (t: Test) => {
+  // Given we have an Auction House and an NFT.
+  const mx = await metaplex();
+  const buyer = await createWallet(mx);
+
+  const nft = await createNft(mx);
+  const { client } = await createAuctionHouse(mx);
+
+  // And we listed that NFT for 1 SOL.
+  const { sellerTradeState } = await client
+    .list({
+      mintAccount: nft.mintAddress,
+      price: sol(1),
+    })
+    .run();
+
+  // And we created a private bid on that NFT for 1 SOL.
+  const { buyerTradeState } = await client
+    .bid({
+      buyer,
+      mintAccount: nft.mintAddress,
+      seller: mx.identity().publicKey,
+      price: sol(1),
+    })
+    .run();
+
+  // When we execute a sale with given listing and bid.
+  await client
+    .executeSale({
+      buyer: toPublicKey(buyer),
+      mintAccount: nft.mintAddress,
+      sellerTradeState,
+      buyerTradeState,
+      price: sol(1),
+      confirmOptions: { skipPreflight: true },
+    })
```

Comment:
If we simply request matching `Listing` and `Bid` models, then the interface should be much simpler and friendlier.

Something like that:

```suggestion
    .executeSale(listing, bid, { confirmOptions: { skipPreflight: true } })
```

Side note: when all of this is done, I was thinking of creating additional "shortcut" operations such as:
- `buy(listing)` which will both create a bid and execute the sale in the same transaction.
- `sell(bid)` which will both create a listing and execute the sale in the same transaction.

What do you think?

Reply:
Hey! I'm still pretty new to the auction house repo. Do you mind giving me an idea on how to implement the ‘buy(listing)` function?

Reply:
@ArjunBhushan The auction house module is still a work in progress. It will be documented in due time. Thank you for your patience. 🌸